frontend/apps/reader/modules/readersearch.lua

```diff
@@ -51,33 +83,95 @@ function ReaderSearch:onShowFulltextSearchInput()
                     text = backward_text,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 1)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = not self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Invalid regular expression!") })
```

Comment:
I'm not sure if we want to shout that? ^_^

Reply:
Do you mean the exclamation mark at the end?

Reply:
Yes, indeed.  By default this has an icon, right? We also have a warning icon, which is often better suited than an exclamation mark in the message, for example. Just something to think about.